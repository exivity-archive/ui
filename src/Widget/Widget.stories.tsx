// @ts-ignore
import { storiesOf } from '@storybook/react'
import faker from 'faker'
import React from 'react'

import { Widget } from '.'
import { Paragraph } from '../Paragraph'

storiesOf('molecules|Widget', module)
  .add('default', () => <Widget title='Title'>
    <Paragraph>{faker.lorem.paragraphs(4)}</Paragraph>
  </Widget>)
