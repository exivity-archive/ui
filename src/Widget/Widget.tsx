import React from 'react'
import styled from 'styled-components'

import { fromTheme, hexToString } from '../utils/styled'
import { Block, BlockProps } from '../Block'

const StyledWidget = styled(Block)<BlockProps>`
  background: white;
  box-shadow: 1px 1px 0 rgba(${fromTheme(theme => hexToString(theme.global.purposes.primary))},0.4);
  box-sizing: border-box;
`

export const Widget = (props: BlockProps) => (
  <StyledWidget borderRadius={3} p={2} {...props}/>
)
