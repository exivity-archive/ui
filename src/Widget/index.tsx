import * as React from 'react'
import styled from 'styled-components'
import { Theme } from '../theme'

const WidgetTitle = styled.h4`
  letter-spacing: -0.025em;
  font-weight: 700;
  font-size: 22px;
  color: ${props => props.theme.colors.blue};
`

interface IWidgetSubTitleProps {
  theme: Theme
}

const WidgetSubTitle = styled.span`
  color: ${(props: IWidgetSubTitleProps) => props.theme.colors.gray};
  font-size: 0.8em;
  padding-left: 10px;
  font-weight: 500;
`

interface IWidget {
  className?: string
  children: React.ReactNode
  title?: string
  subTitle?: string
  theme: Theme
}

const Widget: React.FC<IWidget> = ({ className, children, title, subTitle }) => (
  <div className={className}>
    {title &&
    <WidgetTitle>
      {title}
      {subTitle && <WidgetSubTitle>{subTitle}</WidgetSubTitle>}
    </WidgetTitle>}
    {children}
  </div>
)

export default styled(Widget)`
  background: white;
  box-shadow: 2px 2px 0 lightblue;
  padding: 20px;
  padding-bottom: 100px;
  position: relative;
  box-sizing: border-box;
  border-radius: 3px;
  width: 100%;
`
