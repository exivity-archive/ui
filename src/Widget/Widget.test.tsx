import React from 'react'
import { Widget } from './Widget'
import { render, cleanup } from '@testing-library/react'

afterEach(cleanup)

describe('<Widget/>', () => {
  test('It renders without errors', () => {
    render(<Widget><div /></Widget>)
  })

  test('displayName is Widget', () => {
    expect(Widget.displayName).toBe('Widget')
  })

  test('sticky prop should set position to sticky', () => {
    const { getByText } = render(<Widget sticky>Test</Widget>)
    const widget = getByText('Test')
    expect(widget).toHaveStyleRule('position', 'sticky')
  })

  test('stickyOffset defaults to 0', () => {
    const { getByText } = render(<Widget sticky>Test</Widget>)
    const widget = getByText('Test')
    expect(widget).toHaveStyleRule('top', '0px')
  })

  test('stickyOffset prop sets the top property if sticky prop is true', () => {
    const { getByText } = render(<Widget stickyOffset={10} sticky>Test</Widget>)
    const widget = getByText('Test')
    expect(widget).toHaveStyleRule('top', '10px')
  })

  test('stickyOffset gets ignored if sicky is false', () => {
    const { getByText } = render(<Widget stickyOffset={10}>Test</Widget>)
    const widget = getByText('Test')
    expect(widget).not.toHaveStyleRule('top', '10px')
  })

})
