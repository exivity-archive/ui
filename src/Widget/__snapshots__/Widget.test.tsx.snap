// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef
    subTitle="Widget subTitle"
    title="Widget title"
  >
    Widget
  </ForwardRef>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Widget",
      "forwardedComponent": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "rules": Array [
            "
  background: white;
  box-shadow: 2px 2px 0 lightblue;
  padding: 20px 20px 100px;
  position: relative;
  box-sizing: border-box;
  border-radius: 3px;
  width: 100%;
",
          ],
        },
        "displayName": "Styled(PlainWidget)",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
      "forwardedRef": null,
      "subTitle": "Widget subTitle",
      "title": "Widget title",
    },
    "ref": null,
    "rendered": "Widget",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Widget",
        "forwardedComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  background: white;
  box-shadow: 2px 2px 0 lightblue;
  padding: 20px 20px 100px;
  position: relative;
  box-sizing: border-box;
  border-radius: 3px;
  width: 100%;
",
            ],
          },
          "displayName": "Styled(PlainWidget)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
        "forwardedRef": null,
        "subTitle": "Widget subTitle",
        "title": "Widget title",
      },
      "ref": null,
      "rendered": "Widget",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
